### ISSUE TITLE
[Bug] BlazoredTypeAhead inside BlazorStrap modal: Dropdown disappears when scrollbar is touched

### ISSUE THREAD

**Issue by bancroftway** [Blazored/Typeahead#183]
I have a Blazor Webassembly project running on .Net 5 RC2. I am using a [BlazoredTypeahead ](https://github.com/Blazored/Typeahead) inside a [BlazorStrap ](https://github.com/chanan/BlazorStrap) Modal. When I try to grab the dropdown's scrollbar, the dropdown gets closed. 

I have been unable to determine whether the issue is with BlazoredTypeahead or with BlazorStrap, so I will be cross-posting this item in both the projects.

A reproduction is available here: https://github.com/bancroftway/blazorissuesrepro/tree/typeaheadissues

A video demonstrating the issue is here: https://www.youtube.com/watch?v=m_OftI3jDCQ

**Expected behavior**
Dropdown should not get dismissed when scrollbar is touched.

**Hosting Model (is this issue happening with a certain hosting model?):**
 - Blazor WebAssembly

**Comment by bancroftway** [comment]
As per this [issue ](https://github.com/chanan/BlazorStrap/issues/408)opened with BlazorStrap project, they triaged this issue and they think this issue is happening even without using Blazorstrap. 

So in summary, the issue occurs when using BlazoredTypeahead inside a Bootstrap modal. Please help.

**Comment by chrissainty** [comment]
My apologies for taking a while to respond to this issue @bancroftway. I will look into this as soon as I can and come back to you. 

I've watched your excellent repro video and read the thread from Blazorstrap. It maybe something to do with the z-index of the modal, I'm not sure what or why. I'm just guessing from the video. ANyway I'll get back to you with a something solid ASAP.

**Comment by HAL-NINE-THOUSAND** [comment]
For me this was caused by the blur event on the search input calling ResetControl. 

The only way I could think of resolving it was calling a piece of javascript during the Search method that tracked click events and looked for if the clicked element wasn't a descendent of the search element. When it is it doesn't do anything and when it isn't a descendent it invokes a method on the razor class to trigger the reset.

It works fine but I wonder if there's a non-JS solution

**Comment by chrissainty** [comment]
@HAL-NINE-THOUSAND Would you be able to submit a PR with the changes you made? This one isn't going to get solved without JavaScript, at least as far as I can see, as DOM events are an area where Blazor is very much lacking at the moment.

**Comment by HAL-NINE-THOUSAND** [comment]
@chrissainty 
I've submitted a PR but my original fix was built into a separate solution. I'd used Typeahead as the base for a more focused individual component. Good news is I ported a tweaked version of it to typeahead but bad news is it's not had any IRL testing. I used the repo project in the OP though and the changes resolved the issue. 

Fair warning, it does feel a bit hacky ^^ but I just couldn't think of another way.
